### PR TITLE
TIP-860: Move controller to the right BC

### PR DESCRIPTION
**Pull request by arnolanglade**
**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | yes
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
+<?php
```
Comment (damien-carcel) [inline src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php]: Bad display of GitHub? Or a real issue with git move?

Comment (rouliane) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php]: yeah it's weird, but there are 2 controllers that have been deleted in this PR that could be the original files (based on the name on the corresponding spec on this class)

Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductPdfController.php]: This is a mistake :(

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 2]
```diff
@@ -1,437 +0,0 @@
...
-use Symfony\Component\Validator\Validator\ValidatorInterface;
-
-/**
- * Product controller
- *
- * @author    Julien Sanchez <julien@akeneo.com>
- * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
- */
-class ProductController
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: it is a wrong "git move"

File: src/Pim/Bundle/EnrichBundle/Controller/ProductController.php [hunk 3]
```diff
@@ -1,123 +0,0 @@
...
-use Symfony\Component\Translation\TranslatorInterface;
-
-/**
- * Product Controller
- *
- * @author    Nicolas Dupont <nicolas@akeneo.com>
- * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
- */
-class ProductController extends AbstractListCategoryController
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Pim/Bundle/EnrichBundle/Controller/ProductController.php]: yes it seems it is not used, let's what the CI will say!

File: src/Pim/Bundle/EnrichBundle/Controller/JobExecutionController.php [hunk 4]
```diff
@@ -1,168 +0,0 @@
...
-use Symfony\Component\Translation\TranslatorInterface;
-
-/**
- * Job execution controller
- *
- * @author    Olivier Soulet <olivier.soulet@akeneo.com>
- * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
- */
-class JobExecutionController
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Pim/Bundle/EnrichBundle/Controller/JobExecutionController.php]: yes it seems it is not used, let's what the CI will say!

File: src/Pim/Bundle/EnrichBundle/Controller/GroupController.php [hunk 5]
```diff
@@ -1,112 +0,0 @@
...
-use Symfony\Component\Templating\EngineInterface;
-
-/**
- * Group controller
- *
- * @author    Nicolas Dupont <nicolas@akeneo.com>
- * @copyright 2013 Akeneo SAS (http://www.akeneo.com)
- * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
- */
-class GroupController
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Pim/Bundle/EnrichBundle/Controller/GroupController.php]: yes it seems it is not used, let's what the CI will say!

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/routing/internal_api/product.yml [hunk 6]
```diff
@@ -1,24 +1,100 @@
-pim_enrich_product_model_edit:
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Bundle/Resources/config/routing/internal_api/product.yml]: There are routes about product / product model / product comments in the same file, do you want to split them?

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: This controller has been merge to another one.

File: src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php [hunk 7]
```diff
@@ -2,93 +2,366 @@
...
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  */
 class ProductController
 {
     /** @var ProductRepositoryInterface */
     protected $productRepository;
 
-    /** @var RendererRegistry */
-    protected $rendererRegistry;
+    /** @var CursorableRepositoryInterface */
```
[the remark this answers is not in the document]
Comment (arnolanglade) [reply src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/ProductController.php]: 2 controller have been merge you should have a look to https://github.com/akeneo/pim-community-dev/pull/8876/files/e006be03d0bf43d97fecac76b3c217d738185545#diff-74e44823adf51fa0fa5837a20028db0eL39

File: src/Akeneo/Pim/Enrichment/Bundle/Resources/config/routing/internal_api/product.yml (hunk 6 above)
[the remark this answers is not in the document]
Comment (jjanvier) [reply src/Akeneo/Pim/Enrichment/Bundle/Resources/config/routing/internal_api/product.yml]: as you want
it's OK for me
I just wanted to be sure it was not a mistake

**Pull request merged**